### PR TITLE
[I2C] Harden error management

### PR DESCRIPTION
**Pull request by fpistm**

### CODE REVIEW

File: cores/arduino/stm32/twi.c [hunk 1]
```diff
@@ -723,16 +723,19 @@ i2c_status_e i2c_master_write(i2c_t *obj, uint8_t dev_address,
...
-        } else if (HAL_I2C_GetError(&(obj->handle)) != HAL_I2C_ERROR_NONE) {
+        } else if (err != HAL_I2C_ERROR_NONE) {
           ret = I2C_ERROR;
         }
       }
     }
     /* When Acknowledge failure occurs (Slave don't acknowledge it's address)
        Master restarts communication */
-  } while (HAL_I2C_GetError(&(obj->handle)) == HAL_I2C_ERROR_AF && delta < I2C_TIMEOUT_TICK);
+  } while (((HAL_I2C_GetError(&(obj->handle))&HAL_I2C_ERROR_AF) == HAL_I2C_ERROR_AF)
```
Comment (LMESTM) [inline cores/arduino/stm32/twi.c]: missing spaces around &

File: cores/arduino/stm32/twi.c [hunk 2]
```diff
@@ -784,16 +787,19 @@ i2c_status_e i2c_master_read(i2c_t *obj, uint8_t dev_address, uint8_t *data, uin
...
```
Comment (LMESTM) [inline cores/arduino/stm32/twi.c]: same here

Comment (fpistm) [reply cores/arduino/stm32/twi.c]: done

File: cores/arduino/stm32/twi.c (hunk 1 above)
Comment (fpistm) [reply cores/arduino/stm32/twi.c]: done

**Comment by fpistm** [comment]
> 
> 
> Can you add some details in commit message of what error case this solves ? Was there some timing issue where HAL would report a timeout before this was actually detected locally ?

Some comment added in the commit message

**Pull request merged**